lxd/db/snapshots.mapper.go

```diff
@@ -579,6 +587,7 @@ func (c *ClusterTx) RenameInstanceSnapshot(project string, instance string, name
 }
 
 // DeleteInstanceSnapshot deletes the instance_snapshot matching the given key parameters.
+// generator: instance_snapshot DeleteOne
 func (c *ClusterTx) DeleteInstanceSnapshot(filter InstanceSnapshotFilter) error {
```

Comment:
@masnax not a blocker to this PR, but what (if anything) is stopping us from having the DB generator generate functions that have dedicated arguments for these "DeleteOne" variants? They are already internally at runtime checking that all the required fields are populated, so why not make these fixed arguments instead?

Reply:
This is just a remnant of the initial approach we took, with having everything in the filter.  Although `RenameInstanceSnapshot` isn't a great example as its arguments are actually pulled from the `db:"primary=yes"` tags in the `InstanceSnapshot` struct. 
Using fixed arguments would require including the arguments in the generate statement:`DeleteOne-by-Field1-and-Field2`, which was what the other PR was aiming to do for all of `Delete`, `List`, `Rename`, etc. all at once.

Reply:
OK, how was it done before all the changes began, was it a filter argument (just unvalidated) then?

Reply:
Previously, the generator also pulled from the `db:"primary=yes"` tag, and all other delete methods were manually written.

Reply:
Why cant the arguments for `DeleteInstanceSnapshot` be pulled in the same way as `RenameInstanceSnapshot`?

Reply:
Keeping in mind I dont know what tags are available or what they do :)

Reply:
What I'm trying to say in a roundabout way is that I'd like to get the generated code back to some level of consistency before we start making more changes. It feels like we've started down a particular path, and are now in a half-way house where some functions take a filter (albeit it not really a filter, because all fields in it must be populated or you get a runtime error) and some functions take a strict set of arguments (which effectively is build time validation).